### PR TITLE
Directive to hash a message to create a short digest

### PR DESCRIPTION
**Pull request by nitinmotgi**
Checklist
- [x] Documentation
- [x] Test case
- [x] Added to README.md

### CODE REVIEW

File: wrangler-core/src/main/java/co/cask/wrangler/api/Record.java [hunk 1]
```diff
@@ -229,6 +229,7 @@ public void addOrSet(String name, Object value) {
     }
   }
 
+
```
Comment (russorat) [inline wrangler-core/src/main/java/co/cask/wrangler/api/Record.java]: remove newline

File: wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java [hunk 2]
```diff
@@ -135,6 +138,7 @@ public TextDirectives(String[] directives) {
     formats.put("parse-as-log","parse-as-log <column> <format>");
     formats.put("keep","keep <column>[,<column>]*");
     formats.put("parse-as-hl7", "parse-as-hl7 <column>");
+    formats.put("hash", "hash <column> <algorithm> [replace]");
```
Comment (russorat) [inline wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java:143]: replace -> encode  or something similar?

File: wrangler-core/src/main/java/co/cask/wrangler/steps/MessageHash.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  /**
+   * Checks if the algorithm is the one we support.
+   *
+   * @param algorithm name of the algorithm
+   * @return true if we support, false otherwise.
+   */
+  public static boolean isValid(String algorithm) {
+    if (algorithm != null && algorithms.containsKey(algorithm)) {
```
Comment (russorat) [inline wrangler-core/src/main/java/co/cask/wrangler/steps/MessageHash.java]: ```return (algorithm != null && algorithms.containsKey(algorithm));```

File: wrangler-core/src/test/java/co/cask/wrangler/steps/XMLTest.java [hunk 4]
```diff
@@ -239,5 +239,4 @@ public void testComplexXMLExample() throws Exception {
     Assert.assertTrue(records.size() == 2);
     Assert.assertEquals("Cardigan Sweater", records.get(0).getValue("body.catalog.product.description"));
   }
-
 }
```
Comment (russorat) [inline wrangler-core/src/test/java/co/cask/wrangler/steps/XMLTest.java]: newline

File: wrangler-core/src/test/java/co/cask/wrangler/steps/MessageHashTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    };
+
+    List<Record> records = Arrays.asList(
+      new Record("message1", "This is a very secret message and a digest will be created.")
+    );
+
+    PipelineTest.execute(directives, records);
+  }
+
+}
```
Comment (russorat) [inline wrangler-core/src/test/java/co/cask/wrangler/steps/MessageHashTest.java]: newline

File: wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java (hunk 2 above)
Comment (nitinmotgi) [reply wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java:143]: It's not encoding -- it's hashing.

File: wrangler-core/src/main/java/co/cask/wrangler/api/Record.java (hunk 1 above)
Comment (nitinmotgi) [reply wrangler-core/src/main/java/co/cask/wrangler/api/Record.java]: done!

File: wrangler-core/src/main/java/co/cask/wrangler/steps/MessageHash.java (hunk 3 above)
Comment (nitinmotgi) [reply wrangler-core/src/main/java/co/cask/wrangler/steps/MessageHash.java]: done!

File: wrangler-core/src/test/java/co/cask/wrangler/steps/XMLTest.java (hunk 4 above)
Comment (nitinmotgi) [reply wrangler-core/src/test/java/co/cask/wrangler/steps/XMLTest.java]: done!

File: wrangler-core/src/test/java/co/cask/wrangler/steps/MessageHashTest.java (hunk 5 above)
Comment (nitinmotgi) [reply wrangler-core/src/test/java/co/cask/wrangler/steps/MessageHashTest.java]: done!

File: wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java (hunk 2 above)
Comment (russorat) [reply wrangler-core/src/main/java/co/cask/wrangler/internal/TextDirectives.java:143]: you call it encoding everywhere else. just pick one is all i'm saying.

**Pull request merged**